src/main/java/au/com/addstar/truehardcore/functions/WorldRollback.java

```diff
@@ -135,6 +136,11 @@ public RollbackRequest getNextRequest() {
     public class ProcessNextRequest implements Runnable {
         @Override
         public void run() {
+            // If we're already busy, wait until next cycle
+            if (isQueueBusy()) {
```

Comment:
your testing for queue busy here ....but then you have a bit of code to process ....its plausible that in a multideath situation the time could be just right that it hits this function simultaneously... queue busy could return false for both..
and by the time the aysnc search is called it it would be set true - I think its termed a RACE condition.

Reply:
I could be wrong about this ...I am not sure how this Runnable is called

Reply:
It runs as a sync task on the main thread, so it shouldn't be possible for it to run that twice, but I might rework that to be an atomic lock instead :thinking:

Reply:
Also keep in mind, deaths only add to the queue, they don't process at the same time. The queue processor runs through and handles only one action at a time. But I like the idea of an atomic lock. I'll change it to use that.